implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpStepTimer.java

```diff
@@ -53,10 +53,10 @@ class OtlpStepTimer extends AbstractTimer {
 
     @Override
     protected void recordNonNegative(final long amount, final TimeUnit unit) {
-        final long nanoAmount = (long) TimeUtils.convert(amount, unit, TimeUnit.NANOSECONDS);
+        final long nanoAmount = (long) TimeUtils.convert((double) amount, unit, TimeUnit.NANOSECONDS);
         count.add(1L);
         total.add(nanoAmount);
-        max.record(nanoAmount);
+        max.record((double) nanoAmount);
```

Comment:
I don't really like the chain of conversions here (`long -> double -> integer conversion -> long -> double`) but this PR just makes it explicit so would not change t here. I guess first we need to add `long` overrides to `TimeUtils` to fix this.

@shakuzen What do you think?

Reply:
That's probably a good idea.

Reply:
👍🏼 https://github.com/micrometer-metrics/micrometer/issues/6215